### PR TITLE
feat: Add frame finding [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by rzhao271**
#### Description of Change

Adds the capability to search in specific frames rather than always having to search over the entire webcontents.

CC @deepak1556 

#### Checklist

- [X] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [X] relevant documentation is changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes:

### CODE REVIEW

**Comment by deepak1556** [comment]
Thanks @nornagon, yup the plan is to attempt upstream first, opened this here mainly to cleanup the implementation, adding tests and for visibility. The OP was working on this in a fork where I couldn't give any feedback.

File: patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+           STOP_FIND_ACTION_CLEAR_SELECTION);
+       web_contents_impl->find_request_manager_.release();
+     }
++    for (auto& it : web_contents_impl->frame_find_request_managers_) {
++      it.second->StopFinding(
++          STOP_FIND_ACTION_CLEAR_SELECTION);
++    }
++    web_contents_impl->frame_find_request_managers_.clear();
++  }
++  for (RenderFrameHost* frame : root_frame->GetFramesInSubtree()) {
```
Comment (deepak1556) [inline patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch:246]: what is the loop for ? the previous iteration over `WebContents` also seem to clear any `frame_find_request_managers_` associated with it.

File: patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+@@ -286,18 +288,31 @@ void FindRequestManager::Find(int request_id,
+ }
+ 
+ void FindRequestManager::StopFinding(StopFindAction action) {
+-  for (WebContentsImpl* contents : contents_->GetWebContentsAndAllInner()) {
+-    for (FrameTreeNode* node : contents->GetFrameTree()->Nodes()) {
+-      RenderFrameHostImpl* rfh = node->current_frame_host();
+-      if (!CheckFrame(rfh) || !rfh->IsRenderFrameLive())
++  if (root_frame_) {
++    for (RenderFrameHost* frame: root_frame_->GetFramesInSubtree()) {
```
Comment (deepak1556) [inline patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch:35]: Does `GetFramesInSubtree` include the `root_frame_` ?

File: patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+ 
+     SendFindRequest(request, target_rfh);
+     current_request_ = request;
+@@ -607,6 +624,21 @@ void FindRequestManager::FindInternal(const FindRequest& request) {
+ 
+   // This is an initial find operation.
+   Reset(request);
++
++  if (root_frame_) {
++    for (WebContentsImpl* contents : contents_->GetWebContentsAndAllInner()) {
```
Comment (deepak1556) [inline patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch:93]: Shouldn't this just attach `FrameObserver` to the `webContents` associated with the `root_frame_`, why iterate over all of them ?

File: patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch (hunk 1 above)
Comment (rzhao271) [reply patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch:246]: The previous iteration goes over all the WebContents within `this`, but excludes `this` (so `web_contents_impl != this` when `clear` is called). This iteration goes over all the subframes under `root_frame` that are within the WebContents `this`.

**Comment by deepak1556** [comment]
Closing in favor of https://github.com/electron/electron/pull/28274 that takes an alternative api approach and also added link to upstream issue.

**Pull request closed without merging**